core/frontend/src/components/vehiclesetup/overview/OnboardSensors.vue

```diff
@@ -0,0 +1,190 @@
+        (name) => autopilot_data.parameters.filter((parameter) => parameter.name === name).first(),
+      )
+      const is_at_default_offsets = offset_params.map((param) => param?.value).every((value) => value === 0.0)
+      const is_at_default_scale = scale_params.map((param) => param?.value).every((value) => value === 1.0)
+      return !is_at_default_offsets || !is_at_default_scale
+    },
+    is_water_baro(baro: deviceId) {
+      if (['MS5611', 'KELLERLD'].includes(baro?.deviceName ?? '--') && autopilot.vehicle_type === 'Submarine') {
+        return true
+      }
+      return false
+    },
+    compass_is_calibrated(imu: deviceId) {
+      const compass_number = imu.param.split('COMPASS_DEV_ID')[1]
+      const offset_params_names = [
+        `COMPASS_OFS${compass_number}_X`,
+        `COMPASS_OFS${compass_number}_Y`,
+        `COMPASS_OFS${compass_number}_Z`,
+      ]
+      const diagonal_params_names = [
+        `COMPASS_ODI${compass_number}_X`,
+        `COMPASS_ODI${compass_number}_Y`,
+        `COMPASS_ODI${compass_number}_Z`,
+      ]
+
+      const offset_params = offset_params_names.map(
+        (name) => autopilot_data.parameters.filter((parameter) => parameter.name === name).first(),
+      )
+      const diagonal_params = diagonal_params_names.map(
+        (name) => autopilot_data.parameters.filter((parameter) => parameter.name === name).first(),
+      )
+      if (offset_params.includes(undefined) || diagonal_params.includes(undefined)) {
+        notifier.pushError('PARAM_MISSING', 'Unable to find Compass parameters')
+      }
+
+      const scale_param_name = `COMPASS_SCALE${compass_number}`
+      const scale_param = autopilot_data.parameters.filter((parameter) => parameter.name === scale_param_name).first()
+      const is_at_default_offsets = offset_params.map((param) => param?.value).every((value) => value === 0.0)
+      const is_at_default_diagonals = diagonal_params.map((param) => param?.value).every((value) => value === 0.0)
+      return !is_at_default_offsets || !is_at_default_diagonals || scale_param?.value !== 0.0
+    },
+    baro_status(baro: deviceId) {
+      const radix = baro.param.replace('_DEVID', '')
+      const number = parseInt(radix.replace('BARO', ''), 10)
+      if (this.is_water_baro(baro)) {
+        const value = mavlink_store_get(mavlink, 'VFR_HUD.messageData.message.alt') as number
+        return `${value ?? '--'} m`
+      }
+      const msg = number === 1 ? 'SCALED_PRESSURE' : `SCALED_PRESSURE${number}`
+      const value = mavlink_store_get(mavlink, `${msg}.messageData.message.press_abs`) as number
+      return `${value.toFixed(1) ?? '--'} hPa`
+    },
+    get_pressure_type(baro: deviceId) {
+      if (!this.is_water_baro(baro)) {
+        return 'Barometric'
+      }
+      const spec_gravity_param = autopilot_data.parameters.filter(
+        (parameter) => parameter.name === 'BARO_SPEC_GRAV',
+      ).first()
+
+      return `${printParam(spec_gravity_param)}`
```

Comment:
makes sense.
we could have a scale like:
dead sea ---- sea ---- tears ----- fresh water

Reply:
Lol, nice :-P

If we wanted to use a scale we could try to do densities corresponding to [these water types](https://bluerobotics.com/learn/speed-of-sound-in-water-calculator/#salinity-estimation), but that's perhaps best put into the firmware parameter description as valid options, rather than us only making the ranges available here? Then again, for valid options to work we'd need specific values, whereas the purpose here may be to categorise the range that the currently selected value is within...